third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb

```diff
@@ -0,0 +1,183 @@
+<% autogen_exception -%>
+package google
+<% unless version == 'ga' -%>
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	dataflow "google.golang.org/api/dataflow/v1b3"
+)
+
+// resourceDataflowFlexTemplateJob defines the schema for Dataflow FlexTemplate jobs.
+func resourceDataflowFlexTemplateJob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDataflowFlexTemplateJobCreate,
+		Read:   resourceDataflowFlexTemplateJobRead,
+		Update: resourceDataflowFlexTemplateJobUpdateByReplacement,
+		Delete: resourceDataflowJobDelete,
+		Schema: map[string]*schema.Schema{
+
+			"container_spec_gcs_path": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"on_delete": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"cancel", "drain"}, false),
+				Optional:     true,
+				Default:      "drain",
+			},
+
+			"labels": {
+				Type:             schema.TypeMap,
+				Optional:         true,
+				DiffSuppressFunc: resourceDataflowJobLabelDiffSuppress,
+			},
+
+			"parameters": {
+				Type:     schema.TypeMap,
+				ForceNew: true,
+				Optional: true,
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				// ForceNew applies to both stream and batch jobs
+				ForceNew: true,
+			},
+
+			"job_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// resourceDataflowFlexTemplateJobCreate creates a Flex Template Job from TF code.
+func resourceDataflowFlexTemplateJobCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	region, err := getRegion(d, config)
+	if err != nil {
+		return err
+	}
+
+	request := dataflow.LaunchFlexTemplateRequest{
+		LaunchParameter: &dataflow.LaunchFlexTemplateParameter{
+			ContainerSpecGcsPath: d.Get("container_spec_gcs_path").(string),
+			JobName:              d.Get("name").(string),
+			Parameters:           expandStringMap(d, "parameters"),
+		},
+	}
+	response, err := config.clientDataflow.Projects.Locations.FlexTemplates.Launch(project, region, &request).Do()
+	if err != nil {
+		return err
+	}
+	job := response.Job
+	d.SetId(job.Id)
+	d.Set("job_id", job.Id)
+
+	return resourceDataflowFlexTemplateJobRead(d, meta)
```

Comment:
If a job immediately fails (bad runtime arg, permissions issue) what happens? 
will this resource silently be dropped in [L136-137](https://github.com/GoogleCloudPlatform/magic-modules/pull/3520/files#diff-7411ff0ee82c3c225d802f7f82acdfb1R136-R137)?
Would it be better to block until pipeline enters running state in the create?

Reply:
After initial creation, if the resource reaches an error state before that check, if will register as having failed to create. If it reaches the error state afterwards, it will be removed from Terraform state on a subsequent plan, and Terraform will indicate that it needs to be recreated- not exactly silent, but not exactly noisy either.

Stateful resource management can be done additively, so I'd prefer it doesn't get added in the initial PR. A followup is fine, although completely optional- https://github.com/terraform-providers/terraform-provider-google-beta/blob/master/google-beta/resource_container_cluster.go#L2179-L2203 is the most effective approach I've found for working with states.